grpc-protocol/src/main/java/com/linecorp/armeria/server/grpc/protocol/AbstractUnsafeUnaryGrpcService.java

```diff
@@ -108,20 +110,19 @@ protected final HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req)
     private static CompletableFuture<ByteBuf> deframeMessage(HttpData framed,
                                                              EventLoop eventLoop,
                                                              ByteBufAllocator alloc) {
-        final CompletableFuture<ByteBuf> deframed = new CompletableFuture<>();
-        final ArmeriaMessageDeframerHandler handler = new ArmeriaMessageDeframerHandler(Integer.MAX_VALUE);
-        final HttpDeframer<DeframedMessage> deframer = HttpDeframer.of(handler, alloc);
+        final CompletableFuture<ByteBuf> deframedByteBuf = new CompletableFuture<>();
 
-        StreamMessage.of(framed).subscribe(deframer, eventLoop);
-        deframer.subscribe(singleSubscriber(deframed), eventLoop);
-        return deframed;
+        final ArmeriaMessageDeframer deframer = new ArmeriaMessageDeframer(Integer.MAX_VALUE);
+        HttpRequest.of(DUMMY_HEADERS, framed).decode(deframer, alloc)
+                   .subscribe(singleSubscriber(deframedByteBuf), eventLoop);
```

Comment:
Can we just do
```java
final CompletableFuture<ByteBuf> deframedByteBuf = new CompletableFuture<>();
final ArmeriaMessageDeframer deframer = new ArmeriaMessageDeframer(Integer.MAX_VALUE);
req.decode(deframer, ctx.alloc()).subscribe(singleSubscriber(deframedByteBuf), ctx.eventLoop());
deframedByteBuf.thenCompose(this::handleMessage)...
```
in `doPost`?

Reply:
Oh. Nice idea! I don't have to make a fake request. 🤣